.github/actions/argocd-app-status-action/action.yml

```
@@ -0,0 +1,98 @@
+                    echo "✅ Resource has successful status with the correct image tag: $IMAGE_TAG"
+                    break
+                else
+                    echo "⏳ Resource is still running... Waiting for completion."
+                fi
+            else
+                echo "🚨 Image tag mismatch! Expected: $TARGET_IMAGE_TAG, Found: $IMAGE_TAG"
+            fi
+
+            # Wait before the next check
+            sleep 10
+        done
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)